src/bam/record.jl

```diff
@@ -20,7 +20,7 @@ mutable struct Record <: XAMRecord
     next_refid::Int32
     next_pos::Int32
     tlen::Int32
-    # variable length data
+    # This vector never has unused bytes at the end
```

Comment:
This comment may be confusing outside the context of the previous design, maybe

```suggestion
    # This vector used to have unused bytes at the end, but no longer does
```

or something?